lib/XMLHttpRequest.js

```diff
@@ -391,7 +452,61 @@ function XMLHttpRequest(opts) {
     }
 
     // Load files off the local filesystem (file://)
+    // or data from Data URI (data:)
     if (local) {
+      if (datauri) try {
+        // Triming from original url string since 
+        var data = url.href.substr(5, settings.url.length);
+
+        // separator between header and actual data
+        var parts = data.split(",", 2);
+
+        if (parts.length < 2) throw "Invalid URL";
+
+        var base64 = false;
+        // check if header part has base64
+        var data_headers = parts[0].split(";");
+        
+        // ignore first element since it's MIME type
+        for (var i = 1; i < data_headers.length; ++i) {
+          if (data_headers[i].toLowerCase() === "base64") {
+            base64 = true;
+            break;
+          }
+        }
```

Comment:
Wouldn't this be cleaner as `var isBase64 = dataHeaders.some((header) => header.toLowerCase() === "base64")`?

Reply:
You are correct, and the rest of the project is still using ES5 syntax (even though ES2015 is allowed, as the minimum Node.js version is 12). However, you could still use `var isBase64 = dataHeaders.some(function (header) { return header.toLowerCase() === "base64"; })`

Reply:
Sure, i could do that